descriptor.cc uses memset on types without trivial copy-assignment

**What version of protobuf and what language are you using?**
Version: master
Language: C++

**What operating system (Linux, Windows, ...) and version?**

Linux

**What runtime / compiler are you using (e.g., python version or gcc version)**

GCC 8.2

**What did you do?**

`descriptor.cc` kicks up some new warnings under GCC 8:
```
../../protobuf/src/google/protobuf/descriptor.cc: In member function ‘google::protobuf::Symbol google::protobuf::DescriptorPool::NewPlaceholderWithMutexHeld(const string&, google::protobuf::DescriptorPool::PlaceholderType) const’:
../../protobuf/src/google/protobuf/descriptor.cc:3898:58: warning: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘class google::protobuf::EnumDescriptor’ with no trivial copy-assignment; use assignment or value-initialization instead [-Wclass-memaccess]
     memset(placeholder_enum, 0, sizeof(*placeholder_enum));
                                                          ^
In file included from ../../protobuf/src/google/protobuf/message.h:121,
                 from ../../protobuf/src/google/protobuf/descriptor.pb.h:29,
                 from ../../protobuf/src/google/protobuf/descriptor.cc:50:
../../protobuf/src/google/protobuf/descriptor.h:890:26: note: ‘class google::protobuf::EnumDescriptor’ declared here
 class LIBPROTOBUF_EXPORT EnumDescriptor {
                          ^~~~~~~~~~~~~~
../../protobuf/src/google/protobuf/descriptor.cc:3912:60: warning: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘class google::protobuf::EnumValueDescriptor’ with no trivial copy-assignment; use assignment or value-initialization instead [-Wclass-memaccess]
     memset(placeholder_value, 0, sizeof(*placeholder_value));
                                                            ^
In file included from ../../protobuf/src/google/protobuf/message.h:121,
                 from ../../protobuf/src/google/protobuf/descriptor.pb.h:29,
                 from ../../protobuf/src/google/protobuf/descriptor.cc:50:
../../protobuf/src/google/protobuf/descriptor.h:1046:26: note: ‘class google::protobuf::EnumValueDescriptor’ declared here
 class LIBPROTOBUF_EXPORT EnumValueDescriptor {
                          ^~~~~~~~~~~~~~~~~~~
../../protobuf/src/google/protobuf/descriptor.cc:3931:64: warning: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘class google::protobuf::Descriptor’ with no trivial copy-assignment; use assignment or value-initialization instead [-Wclass-memaccess]
     memset(placeholder_message, 0, sizeof(*placeholder_message));
                                                                ^
In file included from ../../protobuf/src/google/protobuf/message.h:121,
                 from ../../protobuf/src/google/protobuf/descriptor.pb.h:29,
                 from ../../protobuf/src/google/protobuf/descriptor.cc:50:
../../protobuf/src/google/protobuf/descriptor.h:227:26: note: ‘class google::protobuf::Descriptor’ declared here
 class LIBPROTOBUF_EXPORT Descriptor {
                          ^~~~~~~~~~
../../protobuf/src/google/protobuf/descriptor.cc: In member function ‘google::protobuf::FileDescriptor* google::protobuf::DescriptorPool::NewPlaceholderFileWithMutexHeld(const string&) const’:
../../protobuf/src/google/protobuf/descriptor.cc:3965:46: warning: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘class google::protobuf::FileDescriptor’ with no trivial copy-assignment; use assignment or value-initialization instead [-Wclass-memaccess]
   memset(placeholder, 0, sizeof(*placeholder));
                                              ^
In file included from ../../protobuf/src/google/protobuf/message.h:121,
                 from ../../protobuf/src/google/protobuf/descriptor.pb.h:29,
                 from ../../protobuf/src/google/protobuf/descriptor.cc:50:
../../protobuf/src/google/protobuf/descriptor.h:1289:26: note: ‘class google::protobuf::FileDescriptor’ declared here
 class LIBPROTOBUF_EXPORT FileDescriptor {
                          ^~~~~~~~~~~~~~
```
**What did you expect to see**

No warnings.

--- a/src/google/protobuf/descriptor.cc
+++ b/src/google/protobuf/descriptor.cc
@@ -3860,7 +3860,7 @@ Symbol DescriptorPool::NewPlaceholderWithMutexHeld(
     placeholder_file->enum_types_ = tables_->AllocateArray<EnumDescriptor>(1);
 
     EnumDescriptor* placeholder_enum = &placeholder_file->enum_types_[0];
-    memset(placeholder_enum, 0, sizeof(*placeholder_enum));
+    memset(static_cast<void*>(placeholder_enum), 0, sizeof(*placeholder_enum));
 
     placeholder_enum->full_name_ = placeholder_full_name;
     placeholder_enum->name_ = placeholder_name;
@@ -3874,7 +3874,7 @@ Symbol DescriptorPool::NewPlaceholderWithMutexHeld(
     placeholder_enum->values_ = tables_->AllocateArray<EnumValueDescriptor>(1);
 
     EnumValueDescriptor* placeholder_value = &placeholder_enum->values_[0];
-    memset(placeholder_value, 0, sizeof(*placeholder_value));
+    memset(static_cast<void*>(placeholder_value), 0, sizeof(*placeholder_value));
 
     placeholder_value->name_ = tables_->AllocateString("PLACEHOLDER_VALUE");
     // Note that enum value names are siblings of their type, not children.
@@ -3894,7 +3894,7 @@ Symbol DescriptorPool::NewPlaceholderWithMutexHeld(
     placeholder_file->message_types_ = tables_->AllocateArray<Descriptor>(1);
 
     Descriptor* placeholder_message = &placeholder_file->message_types_[0];
-    memset(placeholder_message, 0, sizeof(*placeholder_message));
+    memset(static_cast<void*>(placeholder_message), 0, sizeof(*placeholder_message));
 
     placeholder_message->full_name_ = placeholder_full_name;
     placeholder_message->name_ = placeholder_name;
@@ -3929,7 +3929,7 @@ FileDescriptor* DescriptorPool::NewPlaceholderFileWithMutexHeld(
     mutex_->AssertHeld();
   }
   FileDescriptor* placeholder = tables_->Allocate<FileDescriptor>();
-  memset(placeholder, 0, sizeof(*placeholder));
+  memset(static_cast<void*>(placeholder), 0, sizeof(*placeholder));
 
   placeholder->name_ = tables_->AllocateString(name);
   placeholder->package_ = &internal::GetEmptyString();

